src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncTests.cs

```diff
@@ -4830,5 +4830,47 @@ class C
                 Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "{ await task; }").WithArguments("System.Runtime.CompilerServices.IAsyncStateMachine", "SetStateMachine").WithLocation(62, 37));
         }
 
+        [Fact, WorkItem(16531, "https://github.com/dotnet/roslyn/issues/16531")]
+        public void ArityMismatch()
+        {
+            var source = @"
+using System;
+using System.Threading.Tasks;
+using System.Runtime.CompilerServices;
+
+public class Program
+{
+    public async MyAwesomeType<string> CustomTask() { await Task.Delay(1000); return string.Empty; }
+}
+
+[AsyncMethodBuilder(typeof(CustomAsyncTaskMethodBuilder<,>))]
+public struct MyAwesomeType<T>
+{
+    public T Result { get; set; }
+}
+
+public class CustomAsyncTaskMethodBuilder<T, V>
```

Comment:
>CustomAsyncTaskMethodBuilder<T, V> [](start = 13, length = 34)

It would be good to have a test when a method builder has arity 1, but is nested into a generic type.